### ISSUE TITLE
Multi-Agent Distributed Testing - Agents idling

### ISSUE THREAD

**Issue by joshmackey** [Microsoft/vsts-tasks#8058]
We've got several hundred tests that I'm splitting across 4 VSTS VS2017 hosted agents. I've noticed that if one of the agents finishes its batch of tests early (Using automatic batches of past running time), it idles at `Test run '375' is in 'InProgress' state with ....` until the other test jobs are done.

Is this intentional? Wouldn't it be better to have to finish and free up the concurrentjob for another build to use?

**Comment by acesiddhu** [comment]
Hi Josh
this is by design. We do this because we want to ensure that agents wait till all tests gets executed. if there is a batch of test that failed because of agent issues we queue them to the idle agents to improve resiliency.
I would also suggest to use time based slicing option in the task so that distribution of test are optimized across agents

**Comment by joshmackey** [comment]
I am using the time based slicing option. (I mentioned `Using automatic batches of past running time` above)

Will the automatic batches get better the more I run them? So far it seems that there are too few batches.

**Comment by acesiddhu** [comment]
@joshmackey can you tell us the account name and one of the release/build id where you saw less batches created. We will check our telemetry in details and come back.

**Comment by joshmackey** [comment]
@acesiddhu Can I email that somewhere? I'm also seeing the same issue when I set batch sizes of 60 seconds. One test task gets several minutes worth of tests while the others sit idle.

**Comment by acesiddhu** [comment]
@joshmackey  you can mail here
devops_tools AT Microsoft DOT com

**Comment by joshmackey** [comment]
Got an error saying 
```
Couldn't deliver to the following groups:
--
vs_atpm@microsoft.com, vstlmtcm@microsoft.com
```

**Comment by acesiddhu** [comment]
@joshmackey we got the mail. will reply on that

**Comment by prawalagarwal** [comment]
@joshmackey we have fixed an issue in time based slicing on VsTest task. Please check your runs for change in batching behavior. Thanks!

**Comment by joshmackey** [comment]
Aha, we can't even run builds now as now the GetSources section fails due to git lfs changes (For external git, BitBucket also changed something that makes the BitBucket source not work either)

**Comment by acesiddhu** [comment]
@ericsciple is this a known issue?

**Comment by ericsciple** [comment]
@TingluoHuang ?

**Comment by joshmackey** [comment]
I disabled the GetSources step and wrote some powershell that did the checkout manually. The batching behaviour does indeed seem to be fixed.

Regarding the GetSources step, my colleague wrote up the bug here: https://developercommunity.visualstudio.com/content/problem/320826/bitbucket-cloud-service-connections-cannot-get-sou.html

However, both the External Git AND the build in BitBucket support are not working last I checked.
